### DISCUSSION TITLE
Picture-in-Picture support in chatwoot website widget

### DISCUSSION THREAD

**Discussion by Kaladharshini** [chatwoot/chatwoot#3053] [Q&A]
https://discord.com/channels/647412545203994635/650257251856285707/888727195332280330

For video players, we do want picture-in-picture support.  Attaching the screenshot for example. Assuming we wanted to tweak the widget to support that, where would we start? And is this something chatwoot would also be interested in, so we can contribute upstream?

![chatwoot video](https://user-images.githubusercontent.com/61039762/134003587-a893ca4f-b0b5-410b-9567-f53ab2fcac5c.jpeg)

**Comment by sojan-official** [comment]
@Kaladharshini  Referring to the original PR for interactive messages would be helpful

https://github.com/chatwoot/chatwoot/pull/610

refer to https://github.com/chatwoot/chatwoot/blob/develop/app/javascript/widget/components/AgentMessageBubble.vue for how various message bubbles are rendered. 

One option would be adding a new parameter to card messages where you can provide a video URL instead of the default image URL. this video URL could be rendered in a player which supports picture-in-picture. 

I Will be happy to support you over a PR if you decide to work on this.
